fmriprep.py

```diff
@@ -0,0 +1,235 @@
+    parser.add_argument('-v', '--version', action='store_true',
+                        help="show program's version number and exit")
+
+    # Allow alternative images (semi-developer)
+    parser.add_argument('-i', '--image', metavar='IMG', type=str,
+                        default='poldracklab/fmriprep:latest',
+                        help='image name')
+
+    # Developer patch/shell options
+    g_dev = parser.add_argument_group(
+        'Developer options',
+        'Tools for testing and debugging FMRIPREP')
+    g_dev.add_argument('-f', '--patch-fmriprep', metavar='PATH',
+                       type=os.path.abspath,
+                       help='working fmriprep repository')
+    g_dev.add_argument('-n', '--patch-niworkflows', metavar='PATH',
+                       type=os.path.abspath,
+                       help='working niworkflows repository')
+    g_dev.add_argument('-p', '--patch-nipype', metavar='PATH',
+                       type=os.path.abspath,
+                       help='working nipype repository')
+    g_dev.add_argument('--shell', action='store_true',
+                       help='open shell in image instead of running FMRIPREP')
+    g_dev.add_argument('--scratch', metavar='PATH', type=os.path.abspath,
+                       help='use scratch directory')
+
+    # Capture additional arguments to pass inside container
+    opts, unknown_args = parser.parse_known_args(argv)
+
+    # Stop if no docker / docker fails to run
+    check = check_docker()
+    if check < 1:
+        if opts.version:
+            print('fmriprep wrapper {!s}'.format(__version__))
+        if opts.help:
+            parser.print_help()
+        print("fmriprep: ", end='')
+        if check == -1:
+            print("Could not find docker command... Is it installed?")
+        else:
+            print("Make sure you have permission to run 'docker'")
+        return 1
+
+    # For --help or --version, ask before downloading an image
+    if not check_image(opts.image):
+        resp = 'Y'
+        if opts.version:
+            print('fmriprep wrapper {!s}'.format(__version__))
+        if opts.help:
+            parser.print_help()
+        if opts.version or opts.help:
+            try:
+                resp = input(MISSING.format(opts.image))
+            except KeyboardInterrupt:
+                print()
+                return 1
+        if resp not in ('y', 'Y', ''):
+            return 0
+
+    command = ['docker', 'run', '--rm', '-it']
+
+    if opts.patch_fmriprep is not None:
+        command.extend(['-v', ':'.join((opts.patch_fmriprep,
+                                        '/root/src/fmriprep'))])
+    if opts.patch_niworkflows:
+        command.extend(['-v', ':'.join((opts.patch_niworfklows,
+                                        '/src/niworkflows'))])
+    if opts.patch_nipype:
+        command.extend(['-v', ':'.join((opts.patch_nipype,
+                                        '/src/nipype'))])
```

Comment:
Hmm. This does seem to be a problem, because when pinning with `requirements.txt`, the commit is checked out into `/src/nipype`. So this behavior is going to vary based on whether you're developing a branch that requires a pin or not.